core/src/components/input/input.tsx

```diff
@@ -387,6 +418,11 @@ export class Input implements ComponentInterface {
 
     this.value = '';
 
+    this.ionInput.emit({
+      ...ev,
+      data: this.value,
+    } as InputEvent);
```

Comment:
The underlying event is not actually an `InputEvent`. We should probably update the event type information instead of typecasting this. It looks like native can emit `InputEvent | Event`.